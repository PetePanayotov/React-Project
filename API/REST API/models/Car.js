const mongoose = require('mongoose');

const CarSchema = new mongoose.Schema({

    model: {
        type: String,
        required: true,

    },

    price: {
        type: Number,
        required: true,
       
    },

    imageUrl: {
        type: String,
        required: true
    },

    description : {
        type: String,
        required: true
    },

    likes: [
        {
            type: 'ObjectId',
            required: true,
            ref: 'User'
        }
    ],

    isVipOffer: {
        type: Boolean,
        required: true
    },

    characteristics: {
            required: true,
            type: String
        }
    


});

module.exports = mongoose.model('Car' , CarSchema);