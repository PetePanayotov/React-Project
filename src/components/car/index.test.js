import React from 'react';
import {BrowserRouter , Route} from 'react-router-dom';
import {create , act} from 'react-test-renderer';
import Car from './index';


describe('Car Component' , () => {

    it('Should render catalog page Car article' , () => {

        let root;
        act(() => {
            root = create(
            <BrowserRouter>
                <Route>
                    <Car page="catalog" brand="BMW" model="330 E92" id="1234567" price="30 000" imageUrl="https://google.com/bmw"/>
                </Route>
            </BrowserRouter>
            )
        });

        expect(root.toJSON()).toMatchSnapshot();

    });

    it('Should render admin user page Car article' , () => {

        let root;
        act(() => {
            root = create(
            <BrowserRouter>
                <Route>
                    <Car page="catalog" isAdmin={true} likes="7" brand="BMW" model="330 E92" id="1234567" price="30 000" imageUrl="https://google.com/bmw"/>
                </Route>
            </BrowserRouter>
            )
        });

        expect(root.toJSON()).toMatchSnapshot();

    });

});