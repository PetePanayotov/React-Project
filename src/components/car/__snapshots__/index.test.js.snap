// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car Component Should render admin user page Car article 1`] = `
<div
  className="wrapper"
>
  <section
    className="catalog-imageWrapper"
  >
    <img
      alt="Car Image"
      className="catalog-image"
      src="https://google.com/bmw"
    />
  </section>
  <div>
    <p
      className="paragraph"
    >
      BMW 330 E92
    </p>
    <p
      className="paragraph"
    >
      Price: 
      30 000
       BGN
    </p>
  </div>
  <div
    className="like-link-div"
  >
    <div
      className="car-list-item"
    >
      <a
        className="car-link"
        href="/details?id=1234567"
        onClick={[Function]}
      >
        Findout More
      </a>
    </div>
  </div>
</div>
`;

exports[`Car Component Should render catalog page Car article 1`] = `
<div
  className="wrapper"
>
  <section
    className="catalog-imageWrapper"
  >
    <img
      alt="Car Image"
      className="catalog-image"
      src="https://google.com/bmw"
    />
  </section>
  <div>
    <p
      className="paragraph"
    >
      BMW 330 E92
    </p>
    <p
      className="paragraph"
    >
      Price: 
      30 000
       BGN
    </p>
  </div>
  <div
    className="like-link-div"
  >
    <div
      className="car-list-item"
    >
      <a
        className="car-link"
        href="/details?id=1234567"
        onClick={[Function]}
      >
        Findout More
      </a>
    </div>
  </div>
</div>
`;
